File: 9-palindrome-number/9-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x<0:
            return False
        ori=x
        rev=0
        while x>0:
            rev = rev*10 + x%10
            x = x//10
        if ori==rev:
            return True
        return False

File: 9-palindrome-number/9-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0)return false;
        long n=0;
        int num=x;
        while(true){
            n=n*10+x%10;
            x/=10;
            if(x<=0)break;
        }
        return num==n;
    }
};